libs/langchain/langchain/vectorstores/cratedb/base.py

```diff
@@ -0,0 +1,396 @@
+                            k.lower(): v for k, v in value.items()
+                        }
+                        filter_by_metadata = self.EmbeddingStore.cmetadata[key].in_(
+                            value_case_insensitive[IN]
+                        )
+                        filter_clauses.append(filter_by_metadata)
+                    else:
+                        filter_by_metadata = self.EmbeddingStore.cmetadata[key] == str(
+                            value
+                        )  # type: ignore[assignment]
+                        filter_clauses.append(filter_by_metadata)
+
+                filter_by = sqlalchemy.and_(filter_by, *filter_clauses)
+
+            _type = self.EmbeddingStore
+
+            results: List[Any] = (
+                session.query(  # type: ignore[attr-defined]
+                    self.EmbeddingStore,
+                    # TODO: Original pgvector code uses `self.distance_strategy`.
+                    #       CrateDB currently only supports EUCLIDEAN.
+                    #       self.distance_strategy(embedding).label("distance")
+                    sqlalchemy.literal_column(
+                        f"{self.EmbeddingStore.__tablename__}._score"
+                    ).label("_score"),
+                )
+                .filter(filter_by)
+                # CrateDB applies `KNN_MATCH` within the `WHERE` clause.
+                .filter(
+                    sqlalchemy.func.knn_match(
+                        self.EmbeddingStore.embedding, embedding, k
+                    )
+                )
+                .order_by(sqlalchemy.desc("_score"))
+                .join(
+                    self.CollectionStore,
+                    self.EmbeddingStore.collection_id == self.CollectionStore.uuid,
+                )
+                .limit(k)
+            )
+        return results
+
+    @classmethod
+    def from_texts(  # type: ignore[override]
+        cls: Type[CrateDBVectorSearch],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        collection_name: str = _LANGCHAIN_DEFAULT_COLLECTION_NAME,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        ids: Optional[List[str]] = None,
+        pre_delete_collection: bool = False,
+        **kwargs: Any,
+    ) -> CrateDBVectorSearch:
+        """
+        Return VectorStore initialized from texts and embeddings.
+        Database connection string is required.
+
+        Either pass it as a parameter, or set the CRATEDB_CONNECTION_STRING
+        environment variable.
+
+        Remark: Needs to be overwritten, because CrateDB uses a different
+                DEFAULT_DISTANCE_STRATEGY.
+        """
+        return super().from_texts(  # type: ignore[return-value]
+            texts,
+            embedding,
+            metadatas=metadatas,
+            ids=ids,
+            collection_name=collection_name,
```

Comment:
Re: Filtering by collection: Here, `collection_name` is expected as an input argument to the `from_texts()` method.